nsls2forge_utils/cli.py

```diff
@@ -49,5 +50,45 @@ def main():
         check_package_version(**version_kwargs)
 
 
-if __name__ == '__main__':
-    sys.exit(main())
+def all_feedstocks():
+    parser = argparse.ArgumentParser(
+        description='Retrieve all feedstock repositories from cache or GitHub '
+                    'and write to file.')
+
+    # Set file path
+    parser.add_argument('-f', '--filepath', dest='filepath',
+                        default='names.txt', type=str,
+                        help=('filepath to write feedstock names to '
+                              '(default is names.txt)'))
+
+    # Set cached to true
+    parser.add_argument('-c', '--cached', dest='cached',
+                        action='store_true',
+                        help=('read the names of feedstocks from the cache'))
+
+    # Give GitHub organization name
+    parser.add_argument('-o', '--organization', dest='organization',
+                        default=None, type=str,
+                        help=('GitHub organization to get feedstocks from '
+                              '(must be specified if not cached)'))
+
+    # Give GitHub username and token
+    parser.add_argument('-u', '--username', dest='username',
+                        default=None, type=str,
+                        help=('GitHub username for authentication'))
+    parser.add_argument('-t', '--token', dest='token',
+                        default=None, type=str,
+                        help=('GitHub token for authentication'))
+
+    args = parser.parse_args()
+
+    if not args.cached and args.organization is None:
+        parser.print_help()
```

Comment:
Maybe printing the exact error message and exit would make sense here, so the users can fix the input parameters. See https://docs.python.org/3/library/argparse.html#exiting-methods.